### PR TITLE
refactor(`fungible`): improve `UpdateZRC20WithdrawFee`

### PR DESCRIPTION
**Pull request by lumtis**
# Description

- Add error handling for the contract call
- Replace store iteration with direct verification for zrc20 existence
- Add unit tests for the message

Closes: https://github.com/zeta-chain/node/issues/1108

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go [hunk 1]
```diff
@@ -77,8 +74,8 @@ func (k Keeper) UpdateZRC20WithdrawFee(goCtx context.Context, msg *types.MsgUpda
 	)
 	if err != nil {
 		k.Logger(ctx).Error("failed to emit event", "error", err.Error())
-		return nil, sdkerrors.Wrapf(types.ErrEmitEvent, "failed to emit event (%s)", err.Error())
+		return nil, cosmoserrors.Wrapf(types.ErrEmitEvent, "failed to emit event (%s)", err.Error())
 	}
-	commit()
+
```
Comment (ws4charlie) [inline x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go:79]: The logic was slightly modified vs old version as the new code commits tmpCtx's effect regardless of event emission failure. Needs to be confirmed.

File: x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go [hunk 2]
```diff
@@ -14,55 +15,51 @@ import (
 
 func (k Keeper) UpdateZRC20WithdrawFee(goCtx context.Context, msg *types.MsgUpdateZRC20WithdrawFee) (*types.MsgUpdateZRC20WithdrawFeeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check signer permission
 	if msg.Creator != k.observerKeeper.GetParams(ctx).GetAdminPolicyAccount(zetaObserverTypes.Policy_Type_deploy_fungible_coin) {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Deploy can only be executed by the correct policy account")
+		return nil, cosmoserrors.Wrap(sdkerrors.ErrUnauthorized, "deploy can only be executed by the correct policy account")
```
Comment (ws4charlie) [inline x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go:21]: I love this if it fixed the deprecated 'sdkerrors.Wrap...'

**Review by brewmaster012 — APPROVED** [review]

File: x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go (hunk 2 above)
Comment (lumtis) [reply x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go:21]: Yes, it is the new package

File: x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go (hunk 1 above)
Comment (lumtis) [reply x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go:79]: We use `tmpContext` to ensure the `updateProtocolFlatFee` is atomic. The event emission error is a developer error that must never occur.

Comment (lumtis) [reply x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go:79]: Update: added it back after EmitEvent for consistency as we have this in other messages.

Comment (ws4charlie) [reply x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go:79]: Cool man.

**Review by ws4charlie — APPROVED** [review]
LGTM

**Pull request merged**